app/src/main/java/nl/tudelft/b_b_w/view/ContactsActivity.java

```diff
@@ -0,0 +1,213 @@
+         * Default constructor to initiate the Adapter
+         * @param bc BlockController which is passed on
+         * @param context Context which is passed on
+         */
+        private ContactAdapter(BlockController bc, Context context) {
+            this.context = context;
+            this.blcController = bc;
+        }
+
+        /**
+         *{@inheritDoc}
+         */
+        @Override
+        public Object getItem(int position) {
+            return blcController.getBlocks(ownerName).get(position);
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public long getItemId(int position) {
+            return 0;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public int getCount() {
+            return blcController.getBlocks(ownerName).size();
+        }
+
+
+        /**
+         * Method to get the right image number
+         * @param trust The trust value
+         * @return Image number
+         */
+        private int getImageNo(int trust) {
+
+            if (trust >= 80) return image1;
+            if (trust >= 60) return image2;
+            if (trust >= 40) return image3;
+            if (trust >= 20) return image4;
+            else return image5;
+        }
+
+        /**
+         * Method to create a dialog
+         * @param position Current position of the view
+         * @return The listener
+         */
+        private View.OnClickListener createDialog(final int position) {
+            return new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    AlertDialog.Builder builder = new AlertDialog.Builder(context);
+                    builder.setTitle("Confirm");
+                    builder.setMessage("Are you sure you want to revoke "
+                            + blcController.getBlocks(ownerName).get(position).getOwner()
+                            + " IBAN?");
+                    builder.setPositiveButton("YES", new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int which) {
+                            blcController.revokeBlock(blcController.getBlocks(ownerName).get(position));
+                            notifyDataSetChanged();
+                            dialog.dismiss();
+                        }
+                    });
+                    builder.setNegativeButton("NO", new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            // Do nothing
+                            dialog.dismiss();
+                        }
+                    });
+                    AlertDialog alert = builder.create();
+                    alert.show();
+                }
+            };
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
```

Comment:
I've splitted them up already, but for me it doesn't make sense to split them up even further.